python/GafferSceneUI/InteractiveRenderUI.py

```diff
@@ -36,6 +36,70 @@
 
 import Gaffer
 import GafferScene
+import GafferUI
+
+##########################################################################
+# UI for the state plug that allows setting the state through buttons
+##########################################################################
+
+from collections import namedtuple
+
+ButtonLabels = namedtuple( 'ButtonLabels', 'startButton pauseButton')
+
+class InteractiveRenderingStateWidget( GafferUI.PlugValueWidget ) :
+
+	labelMap = {
+		GafferScene.InteractiveRender.State.Stopped : ButtonLabels('Start', 'Pause'),
+		GafferScene.InteractiveRender.State.Running : ButtonLabels('Stop', 'Pause'),
+		GafferScene.InteractiveRender.State.Paused : ButtonLabels('Continue', 'Paused') }
+
+	def __init__( self, plug, *args, **kwargs) :
+
+		self.plug = plug
+
+		row = GafferUI.ListContainer( GafferUI.ListContainer.Orientation.Horizontal )
+		GafferUI.PlugValueWidget.__init__( self, row, plug )
+
+		with row :
+			self.startButton = GafferUI.Button()
+			self.pauseButton = GafferUI.Button()
+
+			self.startConnection = self.startButton.clickedSignal().connect( Gaffer.WeakMethod( self._startStopCallback ) )
+			self.pauseConnection = self.pauseButton.clickedSignal().connect( Gaffer.WeakMethod( self._pauseCallback ) )
+
+		self._updateButtons()
+
+	def _updateFromPlug( self ) :
+		pass
+
+	def hasLabel( self ) :
+		return True
+
+	def _isRunning( self ) :
+		return self.plug.getValue() == GafferScene.InteractiveRender.State.Running
+
+	def _isStopped( self ) :
+		return self.plug.getValue() == GafferScene.InteractiveRender.State.Stopped
+
+	def _isPaused( self ) :
+		return self.plug.getValue() == GafferScene.InteractiveRender.State.Paused
+
+	def _startStopCallback( self, button ) :
+		if self._isStopped() or self._isPaused() :
+			self.plug.setValue( GafferScene.InteractiveRender.State.Running )
+		else:
+			self.plug.setValue( GafferScene.InteractiveRender.State.Stopped )
+		self._updateButtons()
```

Comment:
Once `_updateFromPlug()` is implemented correctly, this call will be unnecessary. In general, this "change the graph then update my ui" pattern is a bit of an indicator of bugs - the UI should always be updated via a signal of some sort, so that it can respond to externally triggered changes as well as internally triggered ones. The nice side effect of this is that you never need to explicitly make an update after triggering a graph change internally.